wolfssl/wolfcrypt/settings.h

```diff
@@ -314,8 +314,26 @@
         #define WOLFSSL_USER_SETTINGS
     #endif /* WOLFSSL_USER_SETTINGS */
 
-    /* board-specific */
+    /* Board-specific network capability check */
+    #if defined(ARDUINO_SAMD_MKRWIFI1010)    || \
+        defined(ARDUINO_SAMD_NANO_33_IOT)    || \
+        defined(ARDUINO_NANO_RP2040_CONNECT) || \
+        defined(ARDUINO_UNOWIFIR4)           || \
+        defined(ARDUINO_ESP32_DEV)           || \
+        defined(ARDUINO_ESP8266_ESP01)       || \
+        defined(ARDUINO_TEENSY40)            || \
+        defined(ARDUINO_TEENSY41)            || \
+        defined(ESP32) || defined(ESP8266)
+        /* Known to have networking available. */
+    #else
+        #define NO_WOLFSSL_CLIENT
+        #define NO_WOLFSSL_SERVER
+        #define WOLFCRYPT_ONLY
```

Comment:
FYI: WOLFCRYPT_ONLY is all you need if you keep this. Please don't force the other options.

Reply:
Understood. I've removed the entire section for now. 

It may make more sense to keep the board-specific code in the respective ample and not the wolfssl core.